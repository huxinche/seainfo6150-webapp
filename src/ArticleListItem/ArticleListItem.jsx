import React, {useState} from "react";
import PropTypes from "prop-types";
import {Link} from 'react-router-dom';
import ArticleTextToggleButton from '../ArticleTextToggleButton/ArticleTextToggleButton';
import styles from './ArticleListItem.module.css';

const ArticleListItem = (props) => {
  const [show, setShow] = useState(false)
  return (
    <div className={styles.card}>
      <Link to={`/articlelist/${props.slug}`} className={styles.title}>{props.article.title}</Link>
      {show && <>
        <time dateTime={props.article.timeStamp} className={styles.time}>
          {props.article.displayDate}
        </time>
        <p className={styles.text}>{props.article.shortText}</p>
      </>}
      <ArticleTextToggleButton buttonText={show ? 'Show less' : 'Show more'} onClick={() => setShow(!show)}/>
    </div>
  );
};

ArticleListItem.propTypes = {
  article: PropTypes.object.isRequired,
  slug: PropTypes.string.isRequired,
};
export default ArticleListItem;
