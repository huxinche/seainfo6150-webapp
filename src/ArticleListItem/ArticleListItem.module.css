.card{
  border-radius: 6px;
  border: 1px solid #cc0000;
  margin: 16px 8px;
  padding: 12px 14px;
  box-shadow: 0 0 20px rgba(0,0,0,0.6);
  background: linear-gradient(to right, rgba(200,0,0,0.5) 0%, white 20%);
  display: flex;
  flex-direction: column;
}
.main{
  display: flex;
  flex-direction: row;
  align-items: center;
  flex-wrap: wrap;
}
.title{
  color: #cc0000;
  cursor: pointer;
  display: inline-block;
  font-weight: bold;
  font-size: 20px;
  margin-top: 22px;
  min-width: 100px;
  flex: 1;
}
.info{
  flex: 1;
  margin-top: 18px;
}
.time{
  font-weight: bold;
  font-size: 13px;
}

.text{
  font-size: 14px;
  color: black;
  margin-bottom: 20px;
}
