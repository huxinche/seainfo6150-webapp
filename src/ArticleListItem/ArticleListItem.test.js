import React from "react";
import ArticleListItem from "./ArticleListItem";
import article from '../testData.json'
import { fireEvent } from "@testing-library/react";

describe("ArticleListItem tests", () => {
  it("renders correctly", () => {
    const { container } = render(<ArticleListItem article={article} key={article.slug} slug={article.slug} />);
    expect(container).toMatchSnapshot();
  });
  it("renders correctly after button is clicked", () => {
    const { container, getByText } = render(
      <ArticleListItem article={article} key={article.slug} slug={article.slug} />
    );
    const button = getByText("Show more");
    fireEvent.click(button);
    expect(container).toMatchSnapshot();
  });
});
