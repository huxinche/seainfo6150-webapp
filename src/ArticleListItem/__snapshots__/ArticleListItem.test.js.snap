// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleListItem tests renders correctly 1`] = `
<div>
  <div
    class="card"
  >
    <div
      class="main"
    >
      <img
        alt="You can take a pie on a plane"
        class="image"
        src="https://cdn.cnn.com/cnnnext/dam/assets/170321135754-airport-security-line-file-super-169.jpg"
      />
      <a
        class="title"
        href="/articlelist/170321135754-airport-security-line"
      >
        You can take a pie on a plane
      </a>
    </div>
    <div
      class="info"
    />
    <button
      class="button"
    >
      Show more
    </button>
  </div>
</div>
`;

exports[`ArticleListItem tests renders correctly after button is clicked 1`] = `
<div>
  <div
    class="card"
  >
    <div
      class="main"
    >
      <img
        alt="You can take a pie on a plane"
        class="image"
        src="https://cdn.cnn.com/cnnnext/dam/assets/170321135754-airport-security-line-file-super-169.jpg"
      />
      <a
        class="title"
        href="/articlelist/170321135754-airport-security-line"
      >
        You can take a pie on a plane
      </a>
    </div>
    <div
      class="info"
    >
      <time
        class="time"
        datetime="2018-11-21T22:57:55.000Z"
      >
        November 21st 2018, 2:57 pm
      </time>
      <p
        class="text"
      >
        Getting excited to eat your mother's roast turkey, your uncle's stuffing and your sister's famous green bean casserole?
      </p>
    </div>
    <button
      class="button"
    >
      Show less
    </button>
  </div>
</div>
`;
